### DISCUSSION TITLE
Center-when-focused override as a window rule; never-center-at-ends flag

### DISCUSSION THREAD

**Discussion by zetaPRIME** [YaLTeR/niri#874] [Ideas]
Former should be fairly self-explanatory.

As for the latter, having the option for make it so that columns at either end of the flow won't center themselves automatically (unless they're alone on the workspace) would be a nice feel upgrade, especially if you can also override it per-window-rule.

Re: conflicting window rules within the same column, I would suggest having later definitions take precedence the same as happens for individual windows?

**Comment by YaLTeR** [comment]
center-when-focused should be relatively simple, yeah. And doesn't cause column conflicts (you center if any window in the column has that).

never-center-at-ends though would need some reworking of the centering logic to work. Always-center layout has a number of edge cases in the code (e.g. resizing is 2x fast because it goes from both sides at once; the horizontal touchpad gesture snaps to column centers instead of edges; differently sized windows in a column are centered instead of left-justified) and I don't think there's a good way for never-center-at-ends to properly influence (disable) them.

I also think it might be annoying when you have the rightmost window at the edge, then suddenly some window opens to the right, even if it's unfocused, and shifts your view.
